sklearn/ensemble/forest.py

```diff
@@ -417,6 +418,8 @@ def _set_oob_score(self, X, y):
         self.oob_score_ = oob_score / self.n_outputs_
 
     def _validate_y_class_weight(self, y):
+        assert_non_regression_targets(y)
```

Comment:
I thought about is_non_regression_targets, but it is supposed to do nothing at all when it is non regression targets, and raise an exception otherwise. So returning a bool seemed a little superfluous.